dogpile/cache/api.py

```diff
@@ -548,3 +550,22 @@ def set_serialized_multi(
 
         """
         raise NotImplementedError()
+
+
+@dataclass(frozen=True)
+class ValueMetadata:
+    """Holds a value from the cache and associated metadata.
+
+    At the current time, the only metadata is `cached_time` which gives the
+    time at which the value was inserted in the cache (according to
+    `time.time()` on the machine which inserted the value).
+    """
```

Comment:
**Federico Caselli** (CaselIT) wrote:

.. versionadded:: 1.3

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/dogpile.cache/+/5029

Reply:
**Michael Bayer** (zzzeek) wrote:

hm can we use slots for this too?  ill finish this up

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/dogpile.cache/+/5029

Reply:
**Michael Bayer** (zzzeek) wrote:

Done

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/dogpile.cache/+/5029